Lib/test/test_ast.py

```diff
@@ -582,6 +582,19 @@ def test_invalid_identifier(self):
             compile(m, "<test>", "exec")
         self.assertIn("identifier must be of type str", str(cm.exception))
 
+    def test_invalid_constant(self):
+        for invalid_constant in (int, (1, 2, int), frozenset((1, 2, int))):
+            e = ast.Expression(value=ast.Constant(invalid_constant))
+            ast.fix_missing_locations(e)
+            self.assertRaises(
```

Comment:
Oh, it does not look nice. I would use a context manager if you cannot fit the functional form in one line.

Also, the test is passed for different cause than you think. `msg=...` is passed to `compile()`, and since `compile()` does not support the msg keyword argument, it raises a TypeError which is caught by `assertRaises()`. The test is passed, all right. Or not.

Actually you need `assertRaisesRegex()`.

Reply:
Ah, sorry for that. I thought I can use it but looks like this is only for the case of context manager usage. https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises